apps/service_persist/lib/persist/data_storage.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Persist.DataStorage do
+  use Properties, otp_app: :service_persist
+
+  @callback upload(file_path :: String.t(), upload_path :: String.t()) ::
+              {:ok, term} | {:error, term}
+
+  @callback delete(path :: String.t()) :: :ok | {:error, term}
+
+  getter(:impl, default: Persist.DataStorage.S3)
+
+  def upload(file_path, upload_path) do
+    impl().upload(file_path, upload_path)
+  end
+
+  def delete(path) do
+    impl().delete(path)
+  end
+end
+
+defmodule Persist.DataStorage.S3 do
```

Comment:
same question: does it make sense to include these in the same files if you're going to break them out as defined behaviours or just include a single option supported, all in a single module?